src/components/App.js

```diff
@@ -256,6 +261,7 @@ class App extends Component {
             setTimeout(() => {
                 if (this.askForSave()) {
                     this.detachEngines()
+                    try {logger.close()} catch (err) {}
```

Comment:
Please try to eliminate all `try` blocks with `logger` calls by moving them into the `logger` module (if necessary at all).